### PR TITLE
1-1 add unit tests on media refacto

### PR DESCRIPTION
**Pull request by ngilain**

### CODE REVIEW

File: MediaAdminBundle/Tests/FileUtils/AbstractFileUtilsManager.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     * @param string $generatedFile
+     */
+    protected function assertFileCorrectlyGenerated($expectedFile, $generatedFile)
+    {
+        $this->generatedFiles[] = $generatedFile;
+
+        $this->assertFileExists($generatedFile);
+//        $this->assertFileEquals($expectedFile, $generatedFile);
+
+        $expectedImage = new \Imagick($expectedFile);
```
Comment (alavieille) [inline MediaAdminBundle/Tests/FileUtils/AbstractFileUtilsManager.php]: remove \

File: .travis.yml [hunk 2]
```diff
@@ -25,6 +25,8 @@ script: ./bin/phpunit
 addons:
   code_climate:
     repo_token: d1a6f82281cd365fc3b295539d19a932f1bd30a5d1544d6ef3074bd855351db3 
-
+  apt:
```
Comment (nicolasThal) [inline .travis.yml]: to improve the readability, I would suggest to put the `addons.apt` key at the beginning of the file

File: MediaAdminBundle/Tests/EventSubscriber/MediaCreatedSubscriberTest.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    public function testGenerateAlternatives()
+    {
+        $this->subscriber->addMedia($this->event1);
+        $this->subscriber->addMedia($this->event2);
+        $this->subscriber->generateAlternatives();
+
+        Phake::verify($this->fileAlternativesManager)->generateThumbnail($this->media1);
+        Phake::verify($this->fileAlternativesManager)->generateThumbnail($this->media2);
+        Phake::verify($this->fileAlternativesManager)->generateAlternatives($this->media1);
+        Phake::verify($this->fileAlternativesManager)->generateAlternatives($this->media2);
```
Comment (nicolasThal) [inline MediaAdminBundle/Tests/EventSubscriber/MediaCreatedSubscriberTest.php]: You forgot to add `Phake::verify($this->documentManager)->flush()`

File: MediaAdminBundle/Tests/EventSubscriber/MediaDeletedSubscriberTest.php [hunk 4]
```diff
@@ -15,18 +15,24 @@ class MediaDeletedSubscriberTest extends \PHPUnit_Framework_TestCase
...
-        $this->event = Phake::mock('OpenOrchestra\MediaAdmin\Event\MediaEvent');
-        Phake::when($this->event)->getMedia()->thenReturn($this->media);
+        $this->event1 = Phake::mock('OpenOrchestra\MediaAdmin\Event\MediaEvent');
+        Phake::when($this->event1)->getMedia()->thenReturn($this->media1);
+
+        $this->event2 = Phake::mock('OpenOrchestra\MediaAdmin\Event\MediaEvent');
+        Phake::when($this->event2)->getMedia()->thenReturn($this->media2);
 
         $this->fileAlternativesManager = Phake::mock('OpenOrchestra\MediaAdmin\FileAlternatives\FileAlternativesManager');
         $this->subscriber = new MediaDeletedSubscriber($this->fileAlternativesManager);
```
Comment (nicolasThal) [inline MediaAdminBundle/Tests/EventSubscriber/MediaDeletedSubscriberTest.php:38]: You should still use `MediaInterface` in the class `MediaDeletedSubscriber`

File: MediaAdminBundle/Tests/FileUtils/AbstractFileUtilsManager.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+     * 
+     * @param string $expectedFile
+     * @param string $generatedFile
+     */
+    protected function assertFileCorrectlyGenerated($expectedFile, $generatedFile)
+    {
+        $this->generatedFiles[] = $generatedFile;
+
+        $this->assertFileExists($generatedFile);
+//        $this->assertFileEquals($expectedFile, $generatedFile);
```
Comment (nicolasThal) [inline MediaAdminBundle/Tests/FileUtils/AbstractFileUtilsManager.php]: Why do you keep the commented code ?

Comment (ngilain) [inline MediaAdminBundle/Tests/FileUtils/AbstractFileUtilsManager.php]: @nicolasThal : this was made intentionnally to talk about it :) 
Testing the image size is worst than really testing the image itself. But as you have already see it previously, the image generation on travis don't produce the same image as locally (maybe meta-info change).

So do you have a better way to compare 2 images ?

File: MediaAdminBundle/Tests/EventSubscriber/MediaDeletedSubscriberTest.php (hunk 4 above)
Comment (ngilain) [inline MediaAdminBundle/Tests/EventSubscriber/MediaDeletedSubscriberTest.php:38]: @nicolasThal : I don't understand your comment. Where do you think there's a mistake with this interface ?

File: MediaAdminBundle/Tests/FileUtils/AbstractFileUtilsManager.php (hunk 5 above)
Comment (nicolasThal) [inline MediaAdminBundle/Tests/FileUtils/AbstractFileUtilsManager.php]: I remember this issue :).

I have no better way to compare two images at the time..

**Comment by nicolasThal** [comment]
:+1: maybe with the interface use

File: MediaAdminBundle/Tests/FileUtils/AbstractFileUtilsManager.php (hunk 5 above)
Comment (ngilain) [inline MediaAdminBundle/Tests/FileUtils/AbstractFileUtilsManager.php]: Ok I keep the replacement tests and remove the comment one so

**Comment by nicolasThal** [comment]
do not forgot the rebase :)

**Pull request merged**